### PR TITLE
Set up deploy of Enketo servers

### PR DESCRIPTION
**Pull request by dpoirier**

### CODE REVIEW

**Comment by dpoirier** [comment]
@mlavin is this something that you could review?

This started as our Django project template, then removing everything we didn't need, and putting all the new stuff in a single state file, conf/salt/enketo/init.sls.

**Comment by mlavin** [comment]
Sure I can take a look this week.

File: conf/salt/enketo/init.sls [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+      - file: enketo_config
+
+# Add nginx proxy in front
+nginx_ssl_dir:
+  file.directory:
+    - name: /etc/nginx/ssl
+    - user: root
+    - mode: 700
+
+enketo_certificate:
```
Comment (mlavin) [inline conf/salt/enketo/init.sls]: Do we need this self-signed cert? I thought their setup has SSL termination at the ELB?

File: fabfile.py [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+    return yaml.safe_load(result.stdout).keys()
+
+
+def get_project_name():
+    with open(os.path.join(CONF_ROOT, 'pillar', 'project.sls'), 'r') as f:
+        return yaml.safe_load(f)['project_name']
+
+
+@task
+def manage_run(command):
```
Comment (mlavin) [inline fabfile.py]: These aren't really needed :)

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+	echo "WRITE TEST COMMAND"; false
+
+lint-js:
+	# Check JS for any problems
+	# Requires jshint
+	find -name "*.js" -not -path "${STATIC_LIBS_DIR}*" -print0 | xargs -0 jshint
+
+lint: lint-js
+
+$(STATIC_LIBS_DIR):
```
Comment (mlavin) [inline Makefile]: Some of these make commands are really needed either.

**Comment by mlavin** [comment]
One question and some minor (optional) places for removing unused code. Overall this looks good to me.

File: conf/salt/enketo/init.sls (hunk 1 above)
Comment (dpoirier) [inline conf/salt/enketo/init.sls]: I think they had wanted SSL between the ELB and the backend, but self-signed was good enough - just so it's encrypted.

**Comment by dpoirier** [comment]
I've cleaned up the unused code. Was that last comment a "ship it"?

**Comment by mlavin** [comment]
Yes :shipit:

**Pull request merged**